src/DataTypes/DataTypesNumber.cpp

```diff
@@ -2,23 +2,128 @@
 #include <DataTypes/DataTypeFactory.h>
 
 
+#include <Parsers/IAST.h>
+#include <Parsers/ASTLiteral.h>
+
+
 namespace DB
 {
 
+namespace ErrorCodes
+{
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+    extern const int UNEXPECTED_AST_STRUCTURE;
+}
+
+static DataTypePtr createForInt8(const ASTPtr & arguments)
+{
+    if (arguments) {
+        if (arguments->children.size() > 1)
+            throw Exception("INT8 data type family must not have more than one argument - display width", ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+        const auto * argument = arguments->children[0]->as<ASTLiteral>();
+        if (!argument || argument->value.getType() != Field::Types::UInt64 || argument->value.get<UInt64>() == 0)
+            throw Exception("INT8 data type family may have only a number (positive integer) as its argument", ErrorCodes::UNEXPECTED_AST_STRUCTURE);
+    }
+
+    return std::make_shared<DataTypeInt8>();
+}
+
+static DataTypePtr createForInt16(const ASTPtr & arguments)
```

Comment:
Looks like you're adding a lot of copy-paste to this file. This code could be rewritten in a shorter way, for example,

```
template <typename T>
static DataTypePtr createNumericDataType(const ASTPtr & arguments)
{
    if (arguments)
    {
        if (std::is_integral_v<T>)
        {
            if (arguments->children.size() > 1)
                throw Exception(String(TypeName<T>::get()) + " data type family must not have more than one argument - display width", ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
        }
        else
        {
            if (argument->children.size() > 2)
                throw Exception(String(TypeName<T>::get()) + " data type family must not have more than two arguments - total number of digits and number of digits following the decimal point", ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
        }
    }
    return std::make_shared<DataTypeNumber<T>>();
}
```

Reply:
ok, i will do it